cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/upgrade/AwsRdsUpgradeSteps.java

```diff
@@ -0,0 +1,74 @@
+package com.sequenceiq.cloudbreak.cloud.aws.connector.resource.upgrade;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.amazonaws.services.rds.model.DBInstance;
+import com.amazonaws.services.rds.model.DescribeDBInstancesResult;
+import com.sequenceiq.cloudbreak.cloud.aws.common.client.AmazonRdsClient;
+import com.sequenceiq.cloudbreak.cloud.aws.connector.resource.upgrade.operation.AwsRdsUpgradeOperations;
+import com.sequenceiq.cloudbreak.cloud.aws.connector.resource.upgrade.operation.AwsRdsUpgradeValidatorService;
+import com.sequenceiq.cloudbreak.cloud.aws.connector.resource.upgrade.operation.RdsEngineVersion;
+import com.sequenceiq.cloudbreak.cloud.aws.connector.resource.upgrade.operation.RdsInfo;
+import com.sequenceiq.cloudbreak.cloud.aws.connector.resource.upgrade.operation.RdsInstanceStatusesToRdsStateConverter;
+import com.sequenceiq.cloudbreak.cloud.aws.connector.resource.upgrade.operation.RdsState;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.model.DatabaseServer;
+import com.sequenceiq.cloudbreak.common.database.Version;
+
+@Component
+public class AwsRdsUpgradeSteps {
```

Comment:
The main reason was testability: the extracted methods are complex enough with one exception to receive their own tests. Leaving them in AwsRdsUpgradeService would have lead to super bloated tests, to a large number of test cases and when() clauses. It would have been hard to follow those tests, and making sure that all cases are covered.
 
As a result we have a functionality composed set of steps, again composed of a set of operations that are easy to understand and testable.

I know it is possible to use the @VisibleForTesting annotation, but I share the view with others that private methods that need tests on their own should go into a separate class.